app/src/main/java/com/zeapo/pwdstore/UserPreference.kt

```diff
@@ -519,9 +520,32 @@ class UserPreference : AppCompatActivity() {
         startActivityForResult(intent, SET_CUSTOM_XKPWD_DICT)
     }
 
-    @Throws(IOException::class)
+    @Throws(IllegalArgumentException::class, IOException::class)
     private fun copySshKey(uri: Uri) {
-        // TODO: Check if valid SSH Key before import
+        // See metadata from document to validate SSH key
+        contentResolver.query(uri, null, null, null, null, null)?.use { cursor ->
+            val sizeIndex = cursor.getColumnIndex(OpenableColumns.SIZE)
+            // cursor returns only 1 row
+            cursor.moveToFirst()
+            // see file's metadata
+            val fileSize = cursor.getInt(sizeIndex)
+            // We assume that an SSH key's ideal size is > 0 bytes && < 100 kilobytes.
+            if (fileSize > 100000 || fileSize == 0) {
+                throw IllegalArgumentException("Wrong file type selected")
```

Comment:
We should probably extract this error message to a string resource since it is shown to users. I will take care of this.